roles/postgresql_upgrade/tasks/main.yml

```
@@ -0,0 +1,123 @@
+
+- name: Back up PostgreSQL
+  ansible.builtin.command: "{{ postgreql_upgrade_backup_script }}"
+  become: true
+  become_user: postgres
+  changed_when: false
+
+- name: Query for all tables that have WITH OIDS set
+  community.postgresql.postgresql_query:
+    db: "{{ postgressql_upgrade_database }}"
+    query: >
+      SELECT 'ALTER TABLE "' || n.nspname || '"."' || c.relname || '" SET WITHOUT OIDS;' AS "query"
+      FROM pg_catalog.pg_class c
+      LEFT JOIN pg_catalog.pg_namespace n ON n.oid = c.relnamespace
+      WHERE 1=1
+        AND c.relkind = 'r'
+        AND c.relhasoids = true
+        AND n.nspname <> 'pg_catalog'
+      order by n.nspname, c.relname;
+  become: true
+  become_user: postgres
+  register: postgresql_upgrade_tables_with_oids_ddl
+  when: postgresql_upgrade_check_for_oids
+
+- name: Run ALTER statements for tables with OIDS
+  community.postgresql.postgresql_query:
+    db: "{{ postgressql_upgrade_database }}"
+    query: "{{ item.query }}"
+  loop: "{{ postgresql_upgrade_tables_with_oids_ddl.query_result }}"
+  become: true
+  become_user: postgres
+  when: postgresql_upgrade_check_for_oids
+
+- name: Back up PostgreSQL after OIDS edit
+  ansible.builtin.command: "{{ postgreql_upgrade_backup_script }}"
+  become: true
+  become_user: postgres
+  changed_when: false
+  when: postgresql_upgrade_check_for_oids
+
+- name: Disable previous version of PostgreSQL - {{ postgreql_upgrade_current_version }}
+  ansible.builtin.service:
+    name: "postgresql-{{ postgreql_upgrade_current_version }}"
+    state: stopped
+    enabled: false
+
+- name: Install new version of PostgreSQL - {{ postgreql_upgrade_new_version }}
+  ansible.builtin.include_role:
+    name: mirsg.infrastructure.postgresql
+  vars:
+    postgresql_version: "{{ postgreql_upgrade_new_version }}" # noqa var-naming[no-role-prefix]
+    postgresql_create_database: false
+
+- name: Ensure postgres is not running during configuration change
+  ansible.builtin.service:
+    name: "postgresql-{{ postgreql_upgrade_new_version }}"
+    state: stopped
+
+- name: Remove any previous upgrade output
+  ansible.builtin.file:
+    path: "{{ postgresql_upgrade_scripts_dir }}"
+    state: absent
+
+- name: >-
+    Ensure directory for upgrade scripts exists -
+    {{ postgresql_upgrade_scripts_dir }}
+  ansible.builtin.file:
+    path: "{{ postgresql_upgrade_scripts_dir }}"
+    owner: "{{ postgresql_upgrade_postgresql_owner }}"
+    group: "{{ postgresql_upgrade_postgresql_group }}"
+    state: directory
+    mode: "0700"
+
+- name: Run PostgreSQL upgrade
+  ansible.builtin.command:
+    cmd: >
+      /usr/pgsql-{{ postgreql_upgrade_new_version }}/bin/pg_upgrade -d
+      /var/lib/pgsql/{{ postgreql_upgrade_current_version }}/data -D
+      /var/lib/pgsql/{{ postgreql_upgrade_new_version }}/data -b
+      /usr/pgsql-{{ postgreql_upgrade_current_version }}/bin -B
+      /usr/pgsql-{{ postgreql_upgrade_new_version }}/bin -p 10094 -P 5432
```

```suggestion
    cmd: >
      /usr/pgsql-{{ postgreql_upgrade_new_version }}/bin/pg_upgrade
      -d /var/lib/pgsql/{{ postgreql_upgrade_current_version }}/data
      -D /var/lib/pgsql/{{ postgreql_upgrade_new_version }}/data
      -b /usr/pgsql-{{ postgreql_upgrade_current_version }}/bin
      -B /usr/pgsql-{{ postgreql_upgrade_new_version }}/bin
      -p 10094 -P 5432
```
just to make it clearer which value goes with which flag